### DISCUSSION TITLE
Cypress video on ONLY failures not working

### DISCUSSION THREAD

**Discussion by puggerina** [cypress-io/cypress#17761] [General]
I have  "@bahmutov/cypress-extends": "^1.1.0",
and am not using cypress.json am using a base.json instead with values 

"retries": 3,
    "responseTimeout": 120000,
    "defaultCommandTimeout": 120000,
    "pageLoadTimeout": 120000,
    "timeout": 1200000,
    "videoUploadOnPasses":false,
    "viewportWidth": 1280,
    "viewportHeight": 720,
    "chromeWebSecurity": false,
    "reporter": "cypress-teamcity-reporter"

I am not able to generate videos only on failures. Videos are generated on success as well. What am I doing wrong? 
Please help

**Comment by puggerina** [comment]
I added "del": "^6.0.0", and also 
on('after:spec', (spec, results) => {
    if (results.stats.failures === 0 && results.video) {
      // `del()` returns a promise, so it's important to return it to ensure
      // deleting the video is finished before moving on
      return del(result.video)
    }
  })

**Comment by zymotik** [comment]
This is what I am using in our `cypress/plugins/index.js` file to delete all the unneeded video files if the tests pass. This keeps our build artifacts to a minimum.

```javascript
  const storeTestVideo = process.env['CYPRESS_VIDEO_UPLOAD_ON_PASSES'] === undefined 
                                             || process.env['CYPRESS_VIDEO_UPLOAD_ON_PASSES'] == 'true';

  on('after:spec', (spec, results) => {
    if (!storeTestVideo && results.stats.failures === 0 && results.video) {
      fs.unlinkSync(results.video);
    }
  });
```
